src/here-xyz.ts

```diff
@@ -2081,31 +2108,249 @@ function getVirtualSpaceProfiles(title: string, description: string, spaceids: A
         "storage": {
             "id": "virtualspace",
             "params": {
-                virtualspace                
+                virtualspace
             }
         }
     }
 }
 
-function getRuleTaggerProcessorProfile(taggingRules:string) {
-  return  {
-        "id": "rule-tagger",
+function getSchemaProcessorProfile(schema: string) {
+    return {
+        "id": "schema-validator",
+        "eventTypes": ["ModifyFeaturesEvent.request", "ModifySpaceEvent.request"],
         "params": {
-          "taggingRules": {
-            "sport": "$.features[?(@.properties.buildingType==\"stadium\")]"
-          }
+            "schema": schema
         }
-      }
+    }
 }
 
-function getSchemaProcessorProfile(schema:string) {
-   return  { 
-            "id": "schema-validator", 
-            "eventTypes": [ "ModifyFeaturesEvent.request", "ModifySpaceEvent.request" ], 
-            "params": {
-                    "schema": schema 
+function getEmptyRuleTaggerProfile(id:string) {
+    return {
+        "id": id,
+        "eventTypes": ["ModifyFeaturesEvent.request"],
+        "params": {
+            "taggingRules": {
+
+            }
+        }
+    }
+}
+
+function getEmptyRuleTaggerAsyncProcessorProfile() {
+    return getEmptyRuleTaggerProfile("rule-tagger-async");
+}
+
+function getEmptyRuleTaggerProcessorProfile() {
+    return getEmptyRuleTaggerProfile("rule-tagger");
+}
+
+function composeJsonPath(condition: string) {
+    condition = condition.toString();
+    condition = condition.replace(/p\./g, "@.properties.").replace(/f\./g, "@.");
+    let jsonPath: string = "$.features[?(" + condition + ")]";
+    return jsonPath;
+}
+
+function parseJsonPath(jsonPath: string) {
+    let myRegexp = /.\.features\[\?\((.*)\)\]/g;
+    let match: any = myRegexp.exec(jsonPath);
+    let expression = match[1];
+    let condition = expression.replace(/@\.properties\./g, "p.").replace(/@\./g, "f.");
+    return condition;
+}
+
+function isValidTagName(tagName:string) {
+    return tagName && tagName.trim().length > 0 && tagName.indexOf(",") == -1
+}
+
+function isValidRuleExpression(ruleExpression:string) {
+    return ruleExpression && ruleExpression.trim().length > 4
+}
+
+program
+    .command("tagrules <id>")
+    .description("add, remove, view the conditional rules to tag your features automatically, at present all tag rules will be applied synchronously before features are stored ( mode : sync )")
+    .option("--add", "add new entries ( applicable for autotag option only )")
+    .option("--delete", "delete entries ( applicable for autotag option only )")
+    .option("--view", "view entries ( applicable for autotag option only )")
+    // .option("--async", "tag rule will be applied asynchronously after features are written to the storage")
```

Comment:
please add option of update as well

Reply:
hi, the update option is already in the pipeline and is part of another planned commit / story.